### DISCUSSION TITLE
Building fails on mobian: allocation failure

### DISCUSSION THREAD

**Discussion by robert-winkler** [tutao/tutanota#5471] [General]
I'm trying to build the current version of the desktop client 3.112.12 on the PinePhone (arm64) with mobian (debian bullseye).
~~~
- plugin 3 (commonjs) - buildEnd
started building bundles 

<--- Last few GCs --->

[3018:0x55a1cb5580] 502103 ms: Scavenge (reduce) 1488.9 (1521.8) -> 1488.9 (1521.8) MB, 20.6/ 0.0 ms (average mu = 0.278, current mu = 0.249) allocation failure; 

... 

<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
1: 0x7fabec40dc mode::Abort() [/lib/aarch64-linux-gnu/libnode.so.108]

etc.
~~~

Any idea how to fix this?

**Comment by charlag** [comment]
I don't know if building on PinePhone is such a good idea.
Also there are some things (like sqlcipher) that are just not set up for ARM atm so it won't as-is

**Comment by robert-winkler** [comment]
I increased the memory, and it compiles now. However, the sqlite3 module fails. Is there anything I can do? Having the desktop client working on the PinePhone would be awesome.

**Reply by charlag** [reply]
Someone needs to add ARM config and prebuilt things here: https://github.com/tutao/better-sqlite3-sqlcipher
